docs/USAGE-GUIDE.md

```diff
@@ -2287,3 +2287,42 @@ int s2n_early_data_cb_async_impl(struct s2n_connection *conn, struct s2n_offered
 
 To understand the API it may be easiest to see examples in action. s2n-tls's [bin/](https://github.com/aws/s2n-tls/blob/main/bin/) directory
 includes an example client (s2nc) and server (s2nd).
+
+# Unstable Features
+
+This section covers new s2n-tls features that are not yet ready to join the public s2n-tls API. The use of these APIs should not be used for production workloads and 
+they are subject to changes. All public APIs can be found in [api/s2n.h](../api/s2n.h).
```

Comment:
That's what I thought would work best until we are ready to commit to the API. I'm not sure if there is a better alternative or process we follow. I vaguely remembered it was something along these lines

Reply:
If customers are going to use the API, and not just us, then it minimally needs to be labeled "S2N_API" so it's not private, and probably needs to be in a separate header that customers can specifically include.

I think it still needs to be public (with S2N_API) if we only want to use it in the integration tests for now, but it can live in s2n_config.h.

Reply:
Introduce an `api/s2n_unstable.h` file?